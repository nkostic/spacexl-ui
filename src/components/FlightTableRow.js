import React from "react";
import PropTypes from "prop-types";
import { format } from 'date-fns'
import Placeholder from "../assets/placeholder-small.png";
import LinkIco from "../assets/link.png";
import Excerpt from "../lib/Excerpt.js";

const FlightTableRow = ({ flight }) => (
  <div className="columns">
    <div className="column">
      <div className="columns table-row">
        <div className="column first is-1">
          <img
            src={flight.mission_patch_small || Placeholder}
            alt="Mission Patch"
          />
        </div>
        <div className="column">{flight.rocket_name}</div>
        <div className="column">{flight.rocket_type}</div>
        <div className="column">{ format(new Date(flight.launch_date), 'MM/dd/yyyy') }</div>
        <div className="column is-4">{Excerpt(flight.details, 35)}</div>
        <div className="column is-1">{flight.flight_number}</div>
        <div className="column last is-1">
          <a
            href={flight.article_link}
            target="_blank"
            rel="noopener noreferrer"
          >
            <img src={LinkIco} alt="Link Graphic" />
          </a>
        </div>
      </div>
    </div>
  </div>
);

FlightTableRow.propTypes = {
  flight: PropTypes.shape({
    id: PropTypes.number.isRequired,
    rocket_name: PropTypes.string.isRequired,
    rocket_type: PropTypes.string.isRequired,
    launch_date: PropTypes.string.isRequired,
    mission_patch_small: PropTypes.string,
    details: PropTypes.string,
    flight_number: PropTypes.number.isRequired,
    article_link: PropTypes.string.isRequired,
    land_success: PropTypes.bool.isRequired,
    reused: PropTypes.bool.isRequired
  }).isRequired
};

export default FlightTableRow;
