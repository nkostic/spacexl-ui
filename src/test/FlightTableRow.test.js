import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FlightTableRow from '../components/FlightTableRow';

afterEach(cleanup)

test('<FlightTableRow/>', () => {
  const flight = {
    id:1000,
    mission_patch_small:'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
    rocket_name:'Falcon 1',
    rocket_type:'Merlin A',
    launch_date:'2006-03-24T22:30:00.000Z',
    details:null,
    flight_number: 1001,
    article_link: 'https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html'
  }
  const { getByText, container } = render(<FlightTableRow key={flight.id} flight={flight}/>);
  const rocketName = getByText(/Falcon 1/);
  const rocketType = getByText(/Merlin A/);
  const launchDate = getByText(/03\/24\/2006/);
  const details = getByText(/No Details/);
  const flightNumber = getByText(/1001/);
  expect(rocketName).toBeInTheDocument();
  expect(rocketType).toBeInTheDocument();
  expect(launchDate).toBeInTheDocument();
  expect(details).toBeInTheDocument();
  expect(flightNumber).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
  
})
