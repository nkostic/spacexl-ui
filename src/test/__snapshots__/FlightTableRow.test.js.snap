// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlightTableRow/> 1`] = `
<div
  class="columns"
>
  <div
    class="column"
  >
    <div
      class="columns table-row is-centered"
    >
      <div
        class="column is-1 has-text-centered first"
      >
        <figure
          class="image mobile"
        >
          <img
            alt="Mission Patch"
            data-testid="flight-image"
            src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
          />
        </figure>
      </div>
      <div
        class="column"
      >
        <span
          class="mobile-header"
        >
          Rocket Name:
        </span>
        Falcon 1
      </div>
      <div
        class="column"
      >
        <span
          class="mobile-header"
        >
          Rocket Type:
        </span>
        Merlin A
      </div>
      <div
        class="column"
      >
        <span
          class="mobile-header"
        >
          Launch Date:
        </span>
        03/24/2006
      </div>
      <div
        class="column is-4"
      >
        <span
          class="mobile-details"
        />
        <span
          class="is-hidden-mobile"
        >
          No Details
        </span>
      </div>
      <div
        class="column is-1"
      >
        <span
          class="mobile-header"
        >
          ID:
        </span>
        1001
      </div>
      <div
        class="column last is-1"
      >
        <span
          class="mobile-header"
        >
          Article:
        </span>
        <a
          data-testid="flight-link"
          href="https://www.space.com/2196-spacex-inaugural-falcon-1-rocket-lost-launch.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Link Graphic"
            src="link.png"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
